File: recur.py
def max_r(a):
    if len(a) == 1:
        return a[0]
    else:
        print(f"a={a[:-1]}, last={a[-1]}")
        return max(max_r(a[:-1]), a[-1])


a = [4, 2, 7, 8, 1, 9, 0]
print(max_r(a))

File: recur.c
#include <stdio.h>

int max_r(int a[], int n){
    if (n==1){
        return a[0];
    }
    else{
        int temp = max_r(a, n-1);
        return temp>a[n-1]?temp:a[n-1];
    }
}

int main(){
    int a[] ={4, 2, 7, 8, 1, 9, 0};
    int x = max_r(a, 7);
    printf("max=%d\n", x);
}